scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -1727,3 +1727,178 @@ def test_len_and_bool():
     assert rotation_multi_one
     assert rotation_multi
     assert rotation_single
+
+
+def test_from_davenport_single_rotation():
+    axis = [0, 0, 1]
+    quat = Rotation.from_davenport(axis, 'extrinsic', 90, 
+                                   degrees=True).as_quat()
+    expected_quat = np.array([0, 0, 1, 1]) / np.sqrt(2)
+    assert_allclose(quat, expected_quat)
+
+
+def test_davenport_axes_not_orthogonal():
+    ez = [0, 0, 1]
+    ey = [0, 1, 0]
+    ezy = [0, 1, 1]
+    Rotation.from_davenport([ez, ey], 'e', [0, 0])
+    Rotation.from_davenport([ez, ey, ez], 'e', [0, 0, 0])
+    with pytest.raises(ValueError, match="must be orthogonal"):
+        Rotation.from_davenport([ez, ezy], 'e', [0, 0])
+    with pytest.raises(ValueError, match="must be orthogonal"):
+        Rotation.from_davenport([ez, ey, ezy], 'e', [0, 0, 0])
+
+
+def test_as_davenport():
+
+    rnd = np.random.RandomState(0)
```

Comment:
Nitpick: I belive `rng` is a typical name for that (like "random number generator").

Reply:
`rnd` was the variable used in the other functions that were already implemented, it would be best to leave it for consistency, right? If I change all of them it will pollute the PR I think

Reply:
I see.

I'd still use `rng`, but you can keep `rnd` if you like. No need to change it everywhere here.